### DISCUSSION TITLE
How to pass interface to child component in react using typescript

### DISCUSSION THREAD

**Discussion by kuldeepmi7874** [community/community#20665] [Projects and Issues] [answered]
Ok I'm new to using Typescript for react, but I've been programming long enough that this problem makes me feel very dumb...

Literally all I want to do, is define an interface that acts as the type for a property that you pass to a child component.

Like so...

```
interface Foo {
    bar: string;
}

const Child = ({ foo } : Foo) => {

    console.log(foo)

    return <Text>Hello World</Text>

}

const TestPage = () => {

    const fooObject : Foo = { bar: 'something' }

    return <Child foo={fooObject} />

}
```
But I keep seeing this error underlining the foo property I pass in inside the return statement on the TestPage component:

**Comment by holiq** [comment]
You can try like this
```js 
const Child = ({ foo }: { foo: Foo }) =>  {
  // Some code
}
```
Or giving it a named type:
```js
interface ChildProps {
  foo: Foo
}
```
```js
const Child = ({ foo }: ChildProps) => {
  // Some code
}
```
Source from [here](https://stackoverflow.com/a/73000104)

**Comment by chiragmehta900** [comment, marked as answer]
You cannot pass interface through props, as it only exists in build time, not run time. If you want to pass time somewhere you can use generics.


```
export interface AppleProps<T> { /* ... */ }

export function Apple<T>(props: AppleProps<T>) { /* ... */ }

// ...

import Apple from "./";

interface A {
  name: string;
}

const Component = () => {
  return (
     // You can provide generic arguments for react elements
     <Apple<A> />
   );
}
```

**Reply by dohuuhuy** [reply]
vzd

**Reply by dohuuhuy** [reply]
not allowed
